From b9c628501dfd55fc645719fad17210b286e86a03 Mon Sep 17 00:00:00 2001
From: Andreas Blaesius <skate4life@gmx.de>
Date: Tue, 31 Oct 2017 10:01:02 +0100
Subject: [PATCH] Revert "libsecril-shim: Fix CommandInfo struct definition by
 using libbinder"

This reverts commit 236ba788529bf459a6a57b7bd1229ccad018aeda.

Change-Id: I4de872328aebaf3d5139f5f4a4fd227178380ef3
---
 rilsrc/libsecril-shim/Android.mk    | 3 +--
 rilsrc/libsecril-shim/secril-shim.h | 5 ++---
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/rilsrc/libsecril-shim/Android.mk b/rilsrc/libsecril-shim/Android.mk
index ca978cd..176e05e 100644
--- a/rilsrc/libsecril-shim/Android.mk
+++ b/rilsrc/libsecril-shim/Android.mk
@@ -9,8 +9,7 @@ LOCAL_SRC_FILES := \
 LOCAL_SHARED_LIBRARIES := \
 	liblog \
 	libril \
-	libcutils \
-	libbinder
+	libcutils
 
 LOCAL_CFLAGS := -Wall -Werror
 
diff --git a/rilsrc/libsecril-shim/secril-shim.h b/rilsrc/libsecril-shim/secril-shim.h
index e759f61..dd73e13 100644
--- a/rilsrc/libsecril-shim/secril-shim.h
+++ b/rilsrc/libsecril-shim/secril-shim.h
@@ -11,7 +11,6 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <binder/Parcel.h>
 #include <cutils/compiler.h>
 #include <cutils/properties.h>
 #include <sys/cdefs.h>
@@ -54,8 +53,8 @@ typedef struct
 /* TODO: Do we really need to redefine these? They aren't in a header... */
 typedef struct {
     int requestNumber;
-    void (*dispatchFunction) (android::Parcel &p, struct RequestInfo *pRI);
-    int(*responseFunction) (android::Parcel &p, void *response, size_t responselen);
+    void (*dispatchFunction) (void *p, void *pRI);
+    int(*responseFunction) (void *p, void *response, size_t responselen);
 } CommandInfo;
 
 typedef struct RequestInfo {
-- 
2.7.4

